.github/workflows/coverage.yml

```diff
@@ -0,0 +1,162 @@
+      test-chunks: ${{ steps.set-matrix.outputs.chunk-array }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: Check docker hash
+        uses: ./.github/actions/compute-and-check-dependencies-hash
+      - uses: ./.github/actions/build_4C
+        with:
+          cmake-preset: docker_coverage
+          build-targets: full
+          build-directory: ${{ github.workspace }}/build
+          use-ccache: "false"
+      - uses: ./.github/actions/upload_4C_build
+        with:
+          build-directory: ${{ github.workspace }}/build
+          retention-days: 1
+      - uses: ./.github/actions/chunk_test_suite
+        id: set-matrix
+        # Note: We will not use previous runtimes for chunking to avoid running out of storage
+        with:
+          build-directory: ${{ github.workspace }}/build
+          source-directory: ${{ github.workspace }}
+          number-of-chunks: 40 # we need fine grained test chunks to avoid running out of storage
+
+  gcc13_coverage_test:
+    needs: gcc13_coverage_build
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/4c-multiphysics/4c-dependencies-ubuntu24.04:28cfb9d0
+      options: --user root --env OMPI_ALLOW_RUN_AS_ROOT=1 --env OMPI_ALLOW_RUN_AS_ROOT_CONFIRM=1
+    strategy:
+      matrix:
+        test-chunk: ${{fromJson(needs.gcc13_coverage_build.outputs.test-chunks)}}
+    defaults:
+      run:
+        shell: bash
+    steps:
+      - name: Install dependencies
+        run: |
+          apt-get update
+          apt-get upgrade -y
+          apt-get install -y llvm
+      - uses: actions/checkout@v4
+      - name: Check docker hash
+        uses: ./.github/actions/compute-and-check-dependencies-hash
+      - uses: ./.github/actions/download_4C_build
+        with:
+          build-job: gcc13_coverage_build
+      - name: Run all tests and collect raw coverage data
+        run: | # Note: Collect raw coverage data in a file distict for each process
+          cd $GITHUB_WORKSPACE/build
+          export LLVM_PROFILE_FILE="$GITHUB_WORKSPACE/4C-coverage-%m-%p.profraw"
+          time ctest -I $TEST_CHUNK -j `nproc` --output-on-failure  --output-junit $GITHUB_WORKSPACE/gcc13_coverage_test_report-$TEST_CHUNK.xml
+        env:
+          TEST_CHUNK: ${{ matrix.test-chunk }}
+      - name: Collect coverage data
+        run: |
+          cd $GITHUB_WORKSPACE/build
+          llvm-profdata merge -sparse $GITHUB_WORKSPACE/*.profraw -o $GITHUB_WORKSPACE/4C-coverage-data-$TEST_CHUNK.profdata
+        env:
+          TEST_CHUNK: ${{ matrix.test-chunk }}
+      - name: Upload merged coverage data
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ github.job }}-4C-coverage-data-${{ matrix.test-chunk }}.profdata
+          path: |
+            ${{ github.workspace }}/4C-coverage-data-${{ matrix.test-chunk }}.profdata
+          retention-days: 1
+      - name: Upload test report
+        uses: actions/upload-artifact@v4
+        if: success() || failure()
```

Comment:
Can't that be removed? Looks as if this should always be true.

Reply:
It is actually necessary. If we remove it completely, it will be executed only on `success()` (default behavior). We can also not use `always()` as we don't want to execute it if the workflow is canceled. This condition here is equivalent to `!cancelled()`